### DISCUSSION TITLE
feat:  use a medium format (json/yaml/toml) to store icons information for different extension/filename/pattern

### DISCUSSION THREAD

**Discussion by Freed-Wu** [eza-community/eza#457] [Development]
Comes from <https://github.com/ogham/exa/issues/1138>:

It is easier to edit even for those contributors is not familiar with rust <https://github.com/eza-community/eza/blob/afe832b943d328a20f43c6fa69dd8588cf3285ef/src/output/icons.rs>
It is easier to compare the difference between this project's icon and other projects which provide icon display, so we can add icons more easily. Such as <https://github.com/lambdalisue/nerdfont.vim/blob/master/assets/json/extension.json>
It is easier to sort json by some programs like `jq` in CI/CD

Just my 2c, TIA!

**Comment by MartinFillon** [comment]
This needs to be discussed

**Comment by cafkafk** [comment]
My immediate thoughts is that this shouldn't create any performance regression, currently the icons are in a perfect hashmap created at compile time. So maybe this could exist for a user config? But it still would probably be kinda slow.

**Reply by MartinFillon** [reply]
That is why I would prefer, if we want to add better readability to put them in an enum so that we don't loose speed
